examples/flow-counter/containers/App.js

```diff
@@ -0,0 +1,36 @@
+/* @flow */
+
+import { connect } from 'react-redux';
+import Counter from '../components/Counter';
+import type { CounterProps } from '../components/Counter';
+import { increment, decrement, incrementIfOdd, incrementAsync } from '../actions/counter';
+import { AppState, Dispatchable } from '../types';
+import type { Dispatch } from 'redux';
+import type { Component } from 'react';
+
+type StateProps = AppState;
+
+function mapStateToProps(state : AppState) : StateProps {
+  return {
+    counter: state.counter
+  };
+}
+
+type DispatchProps = {
+  increment: () => any,
+  decrement: () => any,
+  incrementIfOdd: () => any,
+  incrementAsync: () => any
+}
+
+function mapDispatchToProps(dispatch: Dispatch<Dispatchable>) : DispatchProps {
+  return {
+    increment: (...args) => dispatch(increment(...args)),
+    decrement: (...args) => dispatch(decrement(...args)),
+    incrementIfOdd: (...args) => dispatch(incrementIfOdd(...args)),
+    incrementAsync: (...args) => dispatch(incrementAsync(...args)),
+  };
+}
+// TODO: Cannot make this to correctly type check that the resulting props match
+// the Counter prop types
+export default connect(mapStateToProps, mapDispatchToProps)(Counter);
```

Comment:
Ah you're right, forgot about those perf optimizations. Will think a bit more about a possible workaround then, or we can just wait for the Flow team to show us the way